### ISSUE TITLE
[Android] Escape key press no longer launches 'backurl'

### ISSUE THREAD

**Issue by blackboxembedded** [organicmaps/organicmaps#5282]
**Describe the issue**
On the Android version only the Escape key no longer launches the provided backurl value.

**Steps to reproduce**
1. Launch OM with URL, ie. om://?backurl=wunderlinq://datagrid
2. Press Escape key to launch the backurl value, ie wunderlinq://datagrid

**Expected behavior**
Escape key press should launch the backurl value if provided with the launch URL.

**System information:**
 - Operating system and its version: Android 13
 - Organic Maps version: 2023.05.08-7-Google
 - Device Model: Pixel 6 Pro

**Comment by biodranik** [comment]
Is it a regression? Did it work in some previous versions?

**Comment by blackboxembedded** [comment]
I was, it was from a PR I submitted a while back.  I'm going to get my OM build environment up to see if I can send a PR.

**Comment by biodranik** [comment]
If it worked in the previous release, then it would be easier to fix.

**Comment by blackboxembedded** [comment]
Another observation, when using a route URL: om://route?backurl=wunderlinq:// it is still working. It loses the backurl when just using to open OM with om://?backurl=wunderlinq://datagrid

**Comment by blackboxembedded** [comment]
I've submitted a PR for a potential fix: https://github.com/organicmaps/organicmaps/pull/5322

**Comment by strump** [comment]
@blackboxembedded did you try to encode `/` symbol as `%2F` ?
Like `om://?backurl=wunderlinq:%2F%2Fdatagrid` .

**Comment by strump** [comment]
One more thing, from [mwm_url.cpp](https://github.com/organicmaps/organicmaps/blob/master/map/mwm_url.cpp#L59) code I see that OrganicMaps expects to see URIs in one of formats:
*  `om://map?...`
*  `om://route?...`
*  `om://search?...`
*  `om://crosshair?...`

You are using URI in a different format: `om://?backurl=wunderlinq://datagrid`. Because of that `Framework.nativeParseAndSetApiUrl(...)` method returns TYPE_INCORRECT result.

Could you please try URI `om://map?backurl=wunderlinq://datagrid` ?

**Comment by blackboxembedded** [comment]
I had tried this, but you then fail here: https://github.com/organicmaps/organicmaps/blob/cd91d5d121807de9b88fad628f9e4a48e4fd955f/android/src/app/organicmaps/intent/Factory.java#LL209C7-L210C22

**Comment by biodranik** [comment]
Did you try `om://map?backurl=wunderlinq%3A%2F%2Fdatagrid` ? What happened in the C++ code?

**Comment by biodranik** [comment]
Let's discuss the use case first. Do you need to simply launch the map app in any previous state with the option to escape from it back to some other app?

**Comment by blackboxembedded** [comment]
That's correct, we just need a way to launch and supply a backurl to OM without interrupting what OM is already doing.

An example use case would be the following:
A user is on their motorbike and has the WunderLINQ app running and viewing their performance data.  

Later the user decides to route themselves to a saved waypoint, the WunderLINQ app uses an om://route URL to launch and have OM route them to that point.

During routing in OM the user decides they want to change to the next song playing on their device, they use ESCAPE to launch the WunderLINQ app and change the song. 

The user now wants to go back to OM for routing. We would then relaunch OM without interrupting the current navigation.

**Comment by biodranik** [comment]
Then we're fixing a wrong problem. API should have a call to show OM in whatever state it was. 

Use om://map for it is incorrect, it implies to show the map instead, say, continue navigation or show search results. 

I thibk it makes sense to introduce explicit `om://` without host/path, but with ?backurl (or any other necessary query parameters) for that case.

**Comment by blackboxembedded** [comment]
Agreed, strump wanted to know if I had tested it 'om://map'.  I have been using 'om://?backurl=wunderlinq://datagrid'.

**Comment by biodranik** [comment]
1. So om:// without params is working now (but is undocumented), right?
2. Does it work if :// in backurl are percent-encoded?

**Comment by blackboxembedded** [comment]
1. iOS yes, Android not without my open PR.
2. Encoding makes no difference.

**Comment by strump** [comment]
I think we should read `backurl` parameter from deep link even if it has format problems. So even for link `om://no-such-page?backurl=wunderlinq://datagrid` OrganicMaps should handle backurl correctly. I think we can merge #5322 but work closer on "Avoid duplicating backurl in java API" #5336
